src/ble/BLEEndPoint.cpp

```diff
@@ -586,6 +586,16 @@ CHIP_ERROR BLEEndPoint::Init(BleLayer * bleLayer, BLE_CONNECTION_OBJECT connObj,
     return CHIP_NO_ERROR;
 }
 
+void BLEEndPoint::AddRef()
+{
+    if (mRefCount == UINT32_MAX)
+    {
+        return;
```

Comment:
_If_ this code is ever reached, that means we will eventually end up with a dead object that still has references to it (i.e. use-after-free), right?

Of course the same would happen on overflow...

The right thing to do is probably to abort if we're at UINT32_MAX (which in practice we won't be) or something.  I see no way to handle it sanely.

Reply:
Similarly, we should add a `VerifyOrDie(mRefCount > 0)` in `Release` as well.

Reply:
Done